src/caching/github-actions-cache.ts

```diff
@@ -345,33 +351,48 @@ export class GitHubActionsCache implements Cache {
    * Otherwise return false.
    */
   private _maybeHandleServiceDown(
-    res: http.IncomingMessage,
+    res: Result<http.IncomingMessage, Error>,
     script: ScriptReference
-  ): boolean {
+  ): res is {ok: false; error: Error} {
     if (this._serviceIsDown) {
       return true;
     }
-    switch (res.statusCode) {
-      default: {
-        return false;
-      }
-      case /* Too Many Requests */ 429: {
-        this._logger.log({
-          script,
-          type: 'info',
-          detail: 'generic',
-          message: `Hit GitHub Actions cache rate limit, caching disabled.`,
-        });
-        break;
-      }
-      case /* Service Unavailable */ 503: {
-        this._logger.log({
-          script,
-          type: 'info',
-          detail: 'generic',
-          message: `GitHub Actions service is unavailable, caching disabled.`,
-        });
-        break;
+    if (!res.ok) {
+      this._logger.log({
+        script,
+        type: 'info',
+        detail: 'generic',
+        message:
+          `Connection error from GitHub Actions service, caching disabled. ` +
+          'Detail: ' +
+          ('code' in res.error
+            ? `${(res.error as Error & {code: string}).code} `
+            : '') +
+          res.error.message,
+      });
+    } else {
+      switch (res.value.statusCode) {
+        default: {
+          return false;
+        }
+        case /* Too Many Requests */ 429: {
+          this._logger.log({
+            script,
+            type: 'info',
+            detail: 'generic',
+            message: `Hit GitHub Actions cache rate limit, caching disabled.`,
+          });
+          break;
```

Comment:
Similar comment to @rictic, but these cases seem to contradict the function's assertion type. Here we know that `res.ok` is truthy, but we're asserting that it's `false`. Can the assertion omit `ok` or otherwise be tightened up?

Reply:
Fixed, see reply to @rictic.

Reply:
Fixed, see reply to @rictic.